src/components/Parameters/Monitoring.tsx

```diff
@@ -0,0 +1,165 @@
+import { Grid,
+  Slider, 
+  TextField, 
+  Typography,
+  Chip, 
+  Button,
+  Stack,
+  Checkbox,
+  FormControlLabel,} from "@mui/material";
+import {
+  Accordion,
+  AccordionDetails,
+  AccordionSummary,
+} from "../ParametersAccordion";
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import { throttle } from "lodash-es";
+import { useMemo, useState } from "react";
+import { useSerialPortRef } from "../../lib/serialContext";
+import { useSerialIntervalSender } from "../../lib/useSerialIntervalSender";
+import { useSerialLineEvent } from "../../lib/useSerialLineEvent";
+import { useParameterSettings } from "../../lib/useParameterSettings";
+import { FocScalar } from "./FocScalar";
+import { MotorMonitorGraph } from "../MotorMonitorGraph";
+
+import {useRef, useEffect} from 'react';
+
+export const Monitoring = (props: {
+  motorKey: string;
+}) => {
+  const fullCommandString = `${props.motorKey}MS`;
+  const serialRef = useSerialPortRef();
+
+  var monitoredVars = "0000000"
+
+  const setCharAt = (str:string,index:number,chr:string) => {
+    if(index > str.length-1) return str;
+    index = index+1;
+    return str.substring(0,index) + chr + str.substring(index+1);
+  }
+  const getCheckboxValue = (id:string) => {
+    var element = document.getElementById(id) as HTMLInputElement | null;
+    if (element == null) return "0";
+    else return element.checked ? "1" : "0";
```

Comment:
@askuric 
Saw ur note in the closed issue about the value lagging behind,
questions, in ur previous attempt with the onChange event, were you reading `e.target.value` or `e.target.checked` ?
you should use the latter, and if that doesn't work then im not sure